### PR TITLE
test: Replace existing NUnit assertions with updated ones to align with NUnit 4 standards.

### PR DESCRIPTION
**Pull request by Dor-bl**
## List of changes

- build(deps): Add NUnit.Analyzers 3.10.0 to test/integration
- chore: Add rule NUnit1032 to recognize Quit as an additional dispose method)
- test: Update existing NUnit assertions with updated ones to align with NUnit 4 standards.
- test: added Assert.Multiple when possible
 
 once this PR is merged we can merge https://github.com/appium/dotnet-client/pull/708 as well.
## Types of changes

What types of changes are you proposing/introducing to the .NET client?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] New feature (non-breaking change which adds value to the project)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)

## Documentation
- [ ] Have you proposed a file change/ PR with appium to update documentation? 
## Integration tests
- [x] Have you provided integration tests to pass against the beta version of appium? (for Bugfix or New feature)

## Details

Please provide more details about changes if it is necessary. If there are new features you can provide code samples showing how they work and possible use cases. Also, you can create [gists](https://gist.github.com) with pasted C# code samples or put them here using markdown. 
About markdown please read [Mastering markdown](https://guides.github.com/features/mastering-markdown/) and [Writing on GitHub](https://docs.github.com/en/get-started/writing-on-github)

### CODE REVIEW

File: test/integration/Android/SettingTest.cs
Comment (KazuCocoa) [inline test/integration/Android/SettingTest.cs:71]: nice

**Review by KazuCocoa — APPROVED** [review]

File: test/integration/Android/SettingTest.cs
Comment (Dor-bl) [reply test/integration/Android/SettingTest.cs:71]: Tell that to NUnit Analyzer 😅

File: test/integration/Android/SearchingTest.cs
Comment (Dor-bl) [inline test/integration/Android/SearchingTest.cs:1]: @KazuCocoa fixed an issue I found before the merge. please approve if needed

**Review by laolubenson — APPROVED** [review]
LGTM! 🎉

File: test/integration/ServerTests/AppiumLocalServerLaunchingTest.cs
Comment (Dor-bl) [inline test/integration/ServerTests/AppiumLocalServerLaunchingTest.cs:1]: @KazuCocoa, @laolubenson Another issue I fixed just now.

**Review by KazuCocoa — APPROVED** [review]

**Pull request merged**
